base/namedtuple.jl

```diff
@@ -209,6 +209,30 @@ function map(f, nt::NamedTuple{names}, nts::NamedTuple...) where names
     NamedTuple{names}(map(f, map(Tuple, (nt, nts...))...))
 end
 
+function _foldl_impl(
+    op,
+    init,
+    itr::Iterators.Pairs{<:Any,<:Any,<:Any,<:NamedTuple{names}},
+) where {names}
+    if @generated
+        ex = :init
+        for (i, n) in enumerate(names)
+            ex = :(op($ex, $(QuoteNode(n)) => vals[$i]))
+        end
+        quote
+            @inline
+            vals = Tuple(values(itr))
+            $ex
+        end
+    else
+        return invoke(_foldl_impl, Tuple{Any,Any,Any}, op, init, itr)
+    end
+end
+
+# dispatch to the generic method with longer named tuple
+_foldl_impl(op, init, itr::Iterators.Pairs{<:Any,<:Any,<:Any,<:NamedTuple{<:Any,<:Any32}}) =
+    invoke(_foldl_impl, Tuple{Any,Any,Any}, op, init, itr)
```

Comment:
Normally we write this now with runtime branches. Eg.

```
    if length(names) < 32
        if @generated
            ....
            quote
                ...
                return vals
            end
        end
    end
    return invoke(_foldl_impl, Tuple{Any,Any,Any}, op, init, itr)
```

Reply:
I'm getting `LoadError(at "compiler/compiler.jl" line 3: LoadError(at "namedtuple.jl" line 212: ErrorException("syntax: invalid syntax (generated)")))` with this approach (53bc87a5dd9918b4d72e602e277a3ab5465cf686) https://build.julialang.org/#/builders/14/builds/5391/steps/3/logs/stdio

Reply:
Partially reverting to use `if @generated` at "top level" 79cf0eca2e9637090520daf2e43f063d3e99ed63 (though I don't know if it's the reason why it didn't work)